File: app/src/main/java/com/weathercock/cleanarchitecture/presentation/view/fragment/AboutFragment.java
package com.weathercock.cleanarchitecture.presentation.view.fragment;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.weathercock.cleanarchitecture.R;
import com.weathercock.cleanarchitecture.databinding.FragmentAboutBinding;
import com.weathercock.cleanarchitecture.presentation.presenter.fragment.AboutFragmentViewModel;

/**
 * About Fragment
 *
 * @author Kensuke
 */
public class AboutFragment extends Fragment {

    private FragmentAboutBinding binding;
    private AboutFragmentViewModel viewModel;

    public static AboutFragment newInstance() {
        AboutFragment fragment = new AboutFragment();
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = new AboutFragmentViewModel(getActivity());
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        if (binding == null) {
            binding = DataBindingUtil.inflate(inflater, R.layout.fragment_about, container, false);
            binding.setViewModel(viewModel);
        }
        return binding.getRoot();
    }

}

File: app/src/main/java/com/kazakago/cleanarchitecture/presentation/view/fragment/AboutFragment.kt
package com.kazakago.cleanarchitecture.presentation.view.fragment

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.kazakago.cleanarchitecture.databinding.FragmentAboutBinding
import com.kazakago.cleanarchitecture.presentation.listener.presenter.fragment.AboutFragmentViewModelListener
import com.kazakago.cleanarchitecture.presentation.presenter.fragment.AboutFragmentViewModel

class AboutFragment : Fragment(), AboutFragmentViewModelListener {

    companion object {
        @JvmStatic
        fun createInstance(): AboutFragment = AboutFragment()
    }

    private val viewModel: AboutFragmentViewModel by lazy { AboutFragmentViewModel(context = activity, listener = this) }
    private lateinit var binding: FragmentAboutBinding

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentAboutBinding.inflate(inflater!!, container, false)
        binding.viewModel = viewModel
        return binding.root
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel.onViewCreated(savedInstanceState = savedInstanceState)
    }

    /* AboutFragmentViewModelListener */

}
